media/js/firefox/whatsnew/whatsnew-54.js

```diff
@@ -2,58 +2,105 @@
-        'gb': 'https://app.adjust.com/2uo1qc?campaign=whats_new&adgroup=Google_Play_Store&creative=badge_gb',
-        'id': 'https://app.adjust.com/2uo1qc?campaign=whats_new&adgroup=Google_Play_Store&creative=badge_id',
-        'in': 'https://app.adjust.com/2uo1qc?campaign=whats_new&adgroup=Google_Play_Store&creative=badge_in',
-        'it': 'https://app.adjust.com/2uo1qc?campaign=whats_new&adgroup=Google_Play_Store&creative=badge_it',
-        'jp': 'https://app.adjust.com/2uo1qc?campaign=whats_new&adgroup=Google_Play_Store&creative=badge_jp',
-        'mx': 'https://app.adjust.com/2uo1qc?campaign=whats_new&adgroup=Google_Play_Store&creative=badge_es-mx',
-        'pl': 'https://app.adjust.com/2uo1qc?campaign=whats_new&adgroup=Google_Play_Store&creative=badge_pl',
-        'ru': 'https://app.adjust.com/2uo1qc?campaign=whats_new&adgroup=Google_Play_Store&creative=badge_ru'
-    };
+            if (newAppStoreURL) {
+                $('#appStoreLink').attr('href', newAppStoreURL);
+            }
 
-    var form = new Mozilla.SendToDevice();
+            if (newPlayStoreURL) {
+                $('#playStoreLink').attr('href', newPlayStoreURL);
+            }
+        },
+        getStoreURL: function(country, platform) {
+            var newURL;
 
-    form.geoCallback = function(countryCode) {
-        // only take action on non-us country codes
-        if (countryCode && countryCode !== 'us') {
-            // make sure the user's countryCode is in our list
-            if (appStoreURLs.hasOwnProperty(countryCode)) {
-                $('#appStoreLink').attr('href', appStoreURLs[countryCode]);
+            // only update non-US
+            if (country && country !== 'us') {
+                switch(platform) {
+                case 'ios':
+                    if (WNP54.appStoreURLs.hasOwnProperty(country)) {
+                        newURL = WNP54.appStoreURLs[country];
+                    }
+
+                    break;
+                case 'android':
+                    if (WNP54.playStoreURLs.hasOwnProperty(country)) {
+                        newURL = WNP54.playStoreURLs[country];
+                    }
+
+                    break;
+                }
             }
 
-            if (playStoreURLs.hasOwnProperty(countryCode)) {
-                $('#playStoreLink').attr('href', playStoreURLs[countryCode]);
+            return newURL;
+        },
+        init: function() {
+            // initialize the form if it exists
+            if (WNP54.sendToDeviceForm.length > 0) {
+                var form = new Mozilla.SendToDevice();
+
+                form.geoCallback = WNP54.updateStoreLinks;
+
+                form.init();
+            } else {
+                // make the request
+                fetch('/country-code.json', {
+                    method: 'GET'
+                }).then(function(response) {
+                    // convert response to JSON
+                    return response.json();
```

Comment:
Hmm, do we need to add a check here that the `fetch` was successful? The MDN docs seem to say that something like a 404 will not throw an error, so it's best to check that `response.ok` is true. There's also an example of checking for a valid content type of `application/json`.

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch